diff --git a/core/java/com/oplus/os/WaveformEffect.java b/core/java/com/oplus/os/WaveformEffect.java
index 8466f78bed1ad..acebf4fa97619 100644
--- a/core/java/com/oplus/os/WaveformEffect.java
+++ b/core/java/com/oplus/os/WaveformEffect.java
@@ -28,10 +28,12 @@ public class WaveformEffect implements Parcelable {
 
     private int mEffectType;
     private boolean mEffectLoop;
+    private boolean mStrengthSettingEnabled;
 
     private WaveformEffect() {
         mEffectType = -1;
         mEffectLoop = false;
+        mStrengthSettingEnabled = false;
     }
 
     public int getEffectType() {
@@ -42,24 +44,32 @@ public class WaveformEffect implements Parcelable {
         return mEffectLoop;
     }
 
+    public boolean getStrengthSettingEnabled() {
+        return mStrengthSettingEnabled;
+    }
+
     public static class Builder {
         private int mEffectType;
         private boolean mEffectLoop;
+        private boolean mStrengthSettingEnabled;
 
         public Builder() {
             mEffectType = -1;
             mEffectLoop = false;
+            mStrengthSettingEnabled = false;
         }
 
         public Builder(WaveformEffect effect) {
             mEffectType = -1;
             mEffectLoop = false;
+            mStrengthSettingEnabled = false;
         }
 
         public WaveformEffect build() {
             WaveformEffect effect = new WaveformEffect();
             effect.mEffectType = mEffectType;
             effect.mEffectLoop = mEffectLoop;
+            effect.mStrengthSettingEnabled = mStrengthSettingEnabled;
             return effect;
         }
 
@@ -72,6 +82,11 @@ public class WaveformEffect implements Parcelable {
             mEffectLoop = loop;
             return this;
         }
+
+        public Builder setStrengthSettingEnabled(boolean enabled) {
+            mStrengthSettingEnabled = enabled;
+            return this;
+        }
     }
 
     @Override
@@ -83,11 +98,13 @@ public class WaveformEffect implements Parcelable {
     public void writeToParcel(Parcel dest, int flags) {
         dest.writeInt(mEffectType);
         dest.writeBoolean(mEffectLoop);
+        dest.writeBoolean(mStrengthSettingEnabled);
     }
 
     private WaveformEffect(Parcel in) {
         mEffectType = in.readInt();
         mEffectLoop = in.readBoolean();
+        mStrengthSettingEnabled = in.readBoolean();
     }
 
     @Override
